studio.py

```
@@ -14,14 +29,201 @@
+
+            platform_name = platform.system().lower()
+
+            if isinstance(self.app, AbstractAndroidFirefox):
+
+                device = ADBAndroid()
+                self.information["ro.product.model"] = device.shell_output(
+                    "getprop ro.product.model"
+                )
+                self.information["ro.build.user"] = device.shell_output(
+                    "getprop ro.build.user"
+                )
+                self.information["ro.build.version.release"] = device.shell_output(
+                    "getprop ro.build.version.release"
+                )
+
+                platform_name = "".join(
+                    e for e in self.information["device_name"] if e.isalnum()
+                )
+
+            for site in sites_json:
+                label = site.get("label")
+                if label:
+                    name = "-".join(
+                        [
+                            self.proxy,
+                            platform_name,
+                            self.app.lower(),
+                            site["domain"],
+                            label,
+                        ]
+                    )
+                else:
+                    name = "-".join(
+                        [self.proxy, platform_name, self.app.lower(), site["domain"]]
+                    )
+                site["path"] = os.path.join(self.path, name, name)
+                site["name"] = name
+
+                site["recording_path"] = "%s.mp" % site["path"]
+                site["json_path"] = "%s.json" % site["path"]
+                site["screen_shot_path"] = "%s.png" % site["path"]
+                site["zip_path"] = os.path.join(self.path, "%s.zip" % site["name"])
+                site["manifest_path"] = os.path.join(
+                    self.path, "%s.manifest" % site["name"]
+                )
+
+                sites.append(site)
+        else:
+            raise Exception("No site file found!!!")
```

This suggests we expect a sites list when recording. If this is the case, let's have the command line parser error out when it's not provided. We should also update the README.

---

This checks that we provide a sites json file. I updated the comment